client/extensions/woocommerce/app/products/product-variation-types-form.js

```diff
@@ -31,14 +31,16 @@ export default class ProductVariationTypesForm extends Component {
 		if ( ! product.attributes ) {
 			this.addType();
 		}
+
+		this.debouncedUpdateName = debounce( this.updateName, 300 );
 	}
 
 	constructor( props ) {
 		super( props );
 
-		this.addType = this.addType.bind( this );
-		this.updateName = this.updateName.bind( this );
-		this.updateValues = this.updateValues.bind( this );
+		this.state = {
+			attributeNames: {},
+		};
```

Comment:
Nice! Since you are not using `props` to initialize `state`, you can avoid the constructor altogether. For example, https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plugins-wpcom/jetpack-plugins-panel.jsx#L57